crates/core/src/physical_plan/stream_table.rs

```diff
@@ -0,0 +1,325 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Generic plans for deferred execution: [`StreamingTableExec`] and [`PartitionStream`]
+
+use std::any::Any;
+use std::sync::Arc;
+
+//use super::{DisplayAs, DisplayFormatType, ExecutionMode, PlanProperties};
+//use crate::display::{display_orderings, ProjectSchemaDisplay};
+//use crate::stream::RecordBatchStreamAdapter;
+//use crate::{ExecutionPlan, Partitioning, SendableRecordBatchStream};
+use arrow::datatypes::SchemaRef;
+use arrow_schema::Schema;
+use datafusion::{
+    common::{internal_err, plan_err, Result},
+    physical_expr::EquivalenceProperties,
+    physical_plan::{
+        display::{display_orderings, ProjectSchemaDisplay},
+        limit::LimitStream,
+        metrics::{BaselineMetrics, MetricsSet},
+        stream::RecordBatchStreamAdapter,
+        streaming::PartitionStream,
+        DisplayAs, DisplayFormatType, ExecutionMode, ExecutionPlan, Partitioning,
+    },
+};
+//use datafusion_execution::TaskContext;
+//use datafusion_physical_expr::{EquivalenceProperties, LexOrdering};
+
+//use crate::limit::LimitStream;
+//use crate::metrics::{BaselineMetrics, ExecutionPlanMetricsSet, MetricsSet};
+use async_trait::async_trait;
+use datafusion::{
+    execution::{SendableRecordBatchStream, TaskContext},
+    physical_expr::LexOrdering,
+    physical_plan::{metrics::ExecutionPlanMetricsSet, PlanProperties},
+};
+use futures::stream::StreamExt;
+use log::debug;
+
+use crate::datasource::kafka::KafkaStreamRead;
+
+pub trait PartitionStreamExt: PartitionStream {
+    fn requires_node_id(&self) -> bool {
+        false
+    }
+
+    fn as_partition_with_node_id(&self) -> Option<&KafkaStreamRead> {
+        None
+    }
+}
+
+/// An [`ExecutionPlan`] for one or more [`PartitionStream`]s.
+///
+/// If your source can be represented as one or more [`PartitionStream`]s, you can
+/// use this struct to implement [`ExecutionPlan`].
```

Comment:
Should add a comment/description as to why DenormalizedStreamingTableExec was created